runtime/android/core/src/org/xwalk/core/XWalkInitializer.java

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.core;
+
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import java.lang.Runnable;
+
+import junit.framework.Assert;
+
+import org.xwalk.core.XWalkLibraryInterface.DecompressionListener;
+import org.xwalk.core.XWalkLibraryInterface.InitializationListener;
+
+/**
+ * XWalkInitializer is a helper to initialize Crosswalk environment without UI interaction. Due to
+ * this limitation, XWalkInitializer works only in embedded mode, or when appropriate Crosswalk
+ * library is already installed on the device in shared mode.
+ */
+public class XWalkInitializer {
```

Comment:
IMPORTANT:
You need to ensure all methods about XWalkInitListener must invoked at UI thread.